src/Apps/W1/Subscription Billing/App/Billing/Codeunits/BillingProposal.Codeunit.al

```diff
@@ -443,6 +444,23 @@ codeunit 8062 "Billing Proposal"
         OnAfterUpdateBillingLineFromSubscriptionLine(BillingLine, ServiceCommitment);
     end;
 
+    local procedure UpdateUsageDataBillingLineNoWhenBillingProposalIsCreated(BillingLine: Record "Billing Line"; SubscriptionLine: Record "Subscription Line")
+    var
+        UsageDataBilling: Record "Usage Data Billing";
+    begin
+        if not SubscriptionLine.IsUsageBasedBillingValid() then
+            exit;
+        UsageDataBilling.FilterOnServiceCommitment(SubscriptionLine);
+        UsageDataBilling.SetRange("Document Type", "Usage Based Billing Doc. Type"::None);
+        UsageDataBilling.SetFilter("Charge Start Date", '<=%1', BillingLine."Billing from");
+        UsageDataBilling.SetFilter("Charge End Date", '>=%1', BillingLine."Billing to");
```

Comment:
There is a function named SetUsageDataBillingFilters. Can we use this one? Or perhaps even IsUsageDataBillingFound? In this case this loop can convert to a two liner.